skimage/filters/ridges.py

```diff
@@ -0,0 +1,651 @@
+
+    References
+    ----------
+    .. [1] Meijering, E., Jacob, M., Sarria, J. C., Steiner, P., Hirling, H.,
+        Unser, M. (2004). Design and validation of a tool for neurite tracing
+        and analysis in fluorescence microscopy images. Cytometry Part A,
+        58(2), 167-176.
+        https://imagescience.org/meijering/publications/download/cyto2004.pdf
+    """
+
+    # Check (sigma) scales
+    sigmas = np.arange(scale_range[0], scale_range[1], scale_step)
+    if np.any(np.asarray(sigmas) < 0.0):
+        raise ValueError("Sigma values less than zero are not valid")
+
+    # Get image dimensions
+    ndim = image.ndim
+
+    # Set parameters
+    if alpha is None:
+        alpha = 1.0 / ndim
+
+    # Invert image to detect bright ridges on dark background
+    if not black_ridges:
+        image = invert(image)
+
+    # Generate empty (n+1)D arrays for storing auxiliary images filtered at
+    # different (sigma) scales
+    filtered_array = np.zeros(sigmas.shape + image.shape)
+
+    # Filtering for all (sigma) scales
+    for i, sigma in enumerate(sigmas):
+
+        # Calculate (sorted) eigenvalues
+        eigenvalues = compute_hessian_eigenvalues(image, sigma, sorting='val')
+
+        if ndim > 1:
+
+            # Set coefficients for scaling eigenvalues
+            coefficients = [alpha] * ndim
+            coefficients[0] = 1
+
+            # Compute auxiliary variables l_i = e_i + sum_{j!=i} alpha * e_j
+            auxiliary = [np.sum([eigenvalues[i] * np.roll(coefficients, j)[i]
+                         for j in range(ndim)], axis=0) for i in range(ndim)]
+
+            # Compute maximum over auxiliary variables
+            auxiliary = np.max(auxiliary, axis=0)
+
+            # Rescale image intensity
+            filtered = np.abs(auxiliary) / np.abs(np.max(auxiliary))
+
+            # Remove background
+            filtered = np.where(filtered > 0, filtered, 0)
+
+            # Store results in (n+1)D matrices
+            filtered_array[i] = filtered
+
+        else:
+
+            # Check image dimensions
+            assert_nD(image, [2, 'more'])
+
+    # Return for every pixel the value of the (sigma) scale with the maximum
+    # output pixel value
+    return np.max(filtered_array, axis=0)
+
+
+def sato(image, scale_range=(1, 10), scale_step=2, black_ridges=True):
+    """
+    Filter an image with the Sato tubeness filter.
+
+    This filter can be used to detect continuous ridges, e.g. tubes,
+    wrinkles, rivers. It can be used to calculate the fraction of the
+    whole image containing such objects.
+
+    Calculates the eigenvectors of the Hessian to compute the similarity of
+    an image region to tubes, according to the method described in _[1].
+
+    Parameters
+    ----------
+    image : (N, M) ndarray
+        Array with input image data.
+    scale_range : 2-tuple of floats, optional
+        The range of sigmas used.
+    scale_step : float, optional
+        Step size between sigmas.
```

Comment:
Yes, that's much easier. Because it deprecates two existing parameters (scale_range and scale_step) in favor of a single new one (I suggest: sigmas) I didn't touch that, but I'll adjust it tomorrow with the corresponding deprecation warnings.

Reply:
Jep, I switched to an iterable of floats now and deprecated the two old scale parameters. Please let me now if the warnings or parameter docstrings should be adjusted.